### DISCUSSION TITLE
IPSec tunnel is stuck in IKE CONNECTING state

### DISCUSSION THREAD

**Discussion by kv-aditya** [strongswan/strongswan#1540] [Q&A]
**System (please complete the following information):**
 - OS: Ubuntu 18.04
 - Kernel version (if applicable): 4.15
 - strongSwan version(s): Linux strongSwan U5.5.3/K4.15.0-54-generic
 - Tested/confirmed with the latest version: no

**Describe the bug**
Configured an IPsec profile with swanctl but the peer is not reachable and so the connection never establishes.  Tried 'unloading' and 'loading' the ipsec profile in this state.  After unloading third time, noticed that the ikev2_init packets are still being sent out.
Took the output of swanctl --list-conn and swanctl --lit-sas.  Though the connection was removed, the sa is stuck in CONNECTING state
root@mypc:/# swanctl --list-conn
root@mypc:/# swanctl --list-sas
aaa: #22, CONNECTING, IKEv2, c6c7cf1a30907e40_i* 0000000000000000_r
  local  '%any' @ 10.42.2.38[500]
  remote '%any' @ 10.42.2.39[500]
  active:  IKE_VENDOR IKE_INIT IKE_NATD IKE_CERT_PRE IKE_AUTH IKE_CERT_POST IKE_CONFIG CHILD_CREATE IKE_AUTH_LIFETIME
Please note that I havent checked the outputs of the commands or the packets sent out before the third attempt.

**Expected behavior**
When an SA is in connecting state, there is no command available in swanctl to terminate it.  In this situation, if the profile is unloaded, the associated SAs also should be stopped/removed.

**Comment by misch42** [comment]
Any log on your side? See: swanctl --log
Any tcpdumps from your side?
Any tcpdump from the other side?
Any logs from the other side?

**Comment by tobiasbrunner** [comment]
> Tried 'unloading' and 'loading' the ipsec profile in this state. After unloading third time, noticed that the ikev2_init packets are still being sent out.

Unless you configured `start_action=start`, unloading a connection will have no effect whatsoever on existing SAs. Neither does changing the config and re-loading it affect SAs that are currently established or getting established.

> When an SA is in connecting state, there is no command available in swanctl to terminate it.

If you want to manually terminate a connection, use the `swanctl --terminate` command.
